keps/sig-storage/1472/README.md

```diff
@@ -0,0 +1,1349 @@
+  details that Kubernetes doesn’t have.
+
+* Nodes are not yet prioritized based on how much storage they have available.
+  This and a way to specify the policy for the prioritization might be
+  added later on.
+
+* Because of that and also for other reasons (capacity changed via
+  operations outside of Kubernetes, like creating or deleting volumes,
+  or expanding the storage), it is expected that pod scheduling may
+  still end up with a node from time to time where volume creation
+  then fails. Rolling back in this case is complicated and outside of
+  the scope of this KEP. For example, a pod might use two persistent
+  volumes, of which one was created and the other not, and then it
+  wouldn’t be obvious whether the existing volume can or should be
+  deleted.
+
+* For persistent volumes that get created independently of a pod
+  nothing changes: it’s still the responsibility of the CSI driver to
+  decide how to create the volume and then communicate back through
+  topology information where pods using that volume need to run.
+  However, a CSI driver may use the capacity information exposed
+  through the proposed API to make its choice.
+
+* Nothing changes for the current CSI ephemeral inline volumes. A [new
+  approach for ephemeral inline
+  volumes](https://github.com/kubernetes/enhancements/issues/1698)
+  will support capacity tracking, based on this proposal here.
+
+### User Stories
+
+#### Ephemeral PMEM volume for Redis or memcached
+
+A [modified Redis server](https://github.com/pmem/redis) and the upstream
+version of [memcached](https://memcached.org/blog/persistent-memory/)
+can use [PMEM](https://pmem.io/) as DRAM replacement with
+higher capacity and lower cost at almost the same performance. When they
+start, all old data is discarded, so an inline ephemeral volume is a
+suitable abstraction for declaring the need for a volume that is
+backed by PMEM and provided by
+[PMEM-CSI](https://github.com/intel/pmem-csi). But PMEM is a resource
+that is local to a node and thus the scheduler has to be aware whether
+enough of it is available on a node before assigning a pod to it.
+
+#### Different LVM configurations
+
+A user may want to choose between higher performance of local disks
+and higher fault tolerance by selecting striping respectively
+mirroring or raid in the storage class parameters of a driver for LVM,
+like for example [TopoLVM](https://github.com/cybozu-go/topolvm).
+
+The maximum size of the resulting volume then depends on the storage
+class and its parameters.
+
+#### Network attached storage
+
+In contrast to local storage, network attached storage can be made
+available on more than just one node. However, for technical reasons
+(high-speed network for data transfer inside a single data center) or
+regulatory reasons (data must only be stored and processed in a single
+jurisdication) availability may still be limited to a subset of the
+nodes in a cluster.
+
+#### Custom schedulers
```

Comment:
When choosing a custom scheduler for a pod through the `PodSpec.SchedulerName`, only the custom scheduler is going to dispatch the pod, so there's no need to disable the feature in the default scheduler. It's still going to be enabled in case that some other pod uses the default scheduler and has volumes, though.